File: analyzer/parse_args.py
import argparse
from config import CONFIG


def parse_arguments():
    parser = argparse.ArgumentParser(description='Linear Regression analysis on given  data')

    parser.add_argument(
        "-n", "--name",
        type=str,
        default=CONFIG["name"],
        help=f'Name of the analysis',
    )

    parser.add_argument(
     "--path",
     type=str,
     default=CONFIG["default_csv"],
     help=f'Path to CSV file (default: {CONFIG["default_csv"]})'
    )

    parser.add_argument(
        "--columns",
        type=str,
        help="Which columns to add for the analysis eg. col1,col2,col3..."
    )

    parser.add_argument(
        "--target",
        type=str,
        help="Target column to predict"
    )
    parser.add_argument(
        "--model",
        type=str,
        default=CONFIG["default_model"]
    )

    parser.add_argument(
        "--clean",
        action="store_false",
        help="Is the data clean"
    )
    parser.add_argument(
        "--output",
        type=str,
        default=CONFIG["output_json"],
        help="Output json file path"
    )

    parser.add_argument(
        "--verbose",
        action="store_false",
        help="Outputs to console in verbose mode"
    )

    parser.add_argument(
        "--visualize",
        action="store_false",
        help="Visualize the output.json"
    )

    parser.add_argument(
        "--port",
        type=str,
        default=CONFIG["default_port"],
        help=f'Serves the output.json on the port. (default: {CONFIG["default_port"]})'
    )

    return parser.parse_args()

File: backend/parse_args.go
package main

import "flag"

type Args struct {
	name           string
	csvInputPath   string
	columns        string
	target         string
	model          string
	clean          bool
	jsonOutputPath string
	verbose        bool
	visualize      bool
	port           string
}

func (app *application) parseArgs() *Args {
	var name string
	var csvInputPath string
	var columns string
	var target string
	var model string
	var clean bool
	var jsonOutputPath string
	var verbose bool
	var visualize bool
	var port string

	flag.StringVar(&name, "name", "lifesat", "Name of the analysis")
	flag.StringVar(&csvInputPath, "path", "../data/lifesat.csv", "Path to csv")
	flag.StringVar(&columns, "columns", "age,gender,income,social_trust,discrimination_index,work_stress,religiosity,social_activity,economic_satisfaction,wellbeing", "Feature columns")
	flag.StringVar(&target, "target", "lifesat", "target column")
	flag.StringVar(&model, "model", "linear_regression", "ML model to use defaults to Multiple Linear Regression")
	flag.BoolVar(&clean, "clean", false, "Is data clean? Default false")
	flag.StringVar(&jsonOutputPath, "output", "../result/result.json", "Results of data analysis and model evaluation")
	flag.BoolVar(&verbose, "verbose", false, "Verbose output")
	flag.BoolVar(&visualize, "visualize", false, "Visualize output")
	flag.StringVar(&port, "port", "8080", "Port to run backend")
	var args = Args{
		name:           name,
		csvInputPath:   csvInputPath,
		columns:        columns,
		target:         target,
		model:          model,
		clean:          clean,
		jsonOutputPath: jsonOutputPath,
		verbose:        verbose,
		visualize:      visualize,
		port:           port,
	}
	return &args
}
